### PR TITLE
Restructure docs.pymc.io, add developer guide (redo #3311)

### PR DESCRIPTION
**Pull request by junpenglao**
Following #3303, this is a further restructure of our website.
- Tutorial page include all our high level API guides (including theano.rst, prod_dist.rst, gp.rst etc)
- renaming of some notebooks (some of them does not have title)
- notebooks might appear under more than 1 categories (if it covers multiple topics)

### CODE REVIEW

**Comment by twiecki** [comment]
Let's try!

**Comment by twiecki** [comment]
Actually we can probably try it on that branch first, were you able to compile it?

**Comment by junpenglao** [comment]
Not yet - still trying to link the rst rendered html to tutorial page.

**Comment by junpenglao** [comment]
It should work now, but just in case - @ColCarroll could you please test this on your website?

**Comment by ColCarroll** [comment]
I actually added a subdomain so it wouldn't be so hacky: https://pymc3.colindcarroll.com

It looks like images are still broken on https://pymc3.colindcarroll.com/nb_tutorials/index.html

File: docs/source/sphinxext/gallery_generator.py [hunk 1]
```diff
@@ -18,6 +18,9 @@
 from matplotlib import image
 
 DOC_SRC = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+DEFAULT_IMG_LOC = os.path.join(
+            os.path.dirname(DOC_SRC), "logos", "PyMC3.png"
```
Comment (ColCarroll) [inline docs/source/sphinxext/gallery_generator.py:22]: somehow this is getting way spread out... i'll look more at how to fix it tomorrow...

Comment (junpenglao) [reply docs/source/sphinxext/gallery_generator.py:22]: Thanks!

**Comment by junpenglao** [comment]
Besides the hiccups of the figure I think it looks nice - excited to have it merge!

**Comment by junpenglao** [comment]
Should be fixed now @ColCarroll

**Comment by ColCarroll** [comment]
Repushed -- it looks good to me!

**Comment by twiecki** [comment]
🤞

**Pull request merged**